src/distributions/weighted_index.rs

```diff
@@ -130,6 +130,72 @@ impl<X: SampleUniform + PartialOrd> WeightedIndex<X> {
         })
     }
 
+    /// Updates all weights by recalculating the index, without changing the number of weights.
+    ///
+    /// **NOTE:** if `weights` contains invalid elements (for example, `f64::NAN` in the case of
```

Comment:
NaN will fail the `w >= &zero` check. What wouldn't be caught is +inf (or a sum of weights which overflows to +inf). Possibly we should add a check for this (`total_weight.is_finite()`).

Despite this comment, the cases which are caught are identical to those of the `new` constructor. Possibly both need updating.